types/index.d.ts

```diff
@@ -0,0 +1,65 @@
+declare module 'svelte-routing' {
+  import 'svelte2tsx/svelte-jsx';
+  import { SvelteComponent, SvelteComponentTyped } from 'svelte';
+  
+  interface RouterProps {
+    basepath?: string;
+    url?: string;
+  }
+
+  interface RouteProps {
+    path?: string;
+    component?: SvelteComponent;
+    [additionalProp: string]: unknown;
+  }
+
+  interface RouteSlots {
+    default: {
+      location: RouteLocation;
+      params: RouteParams;
+    };
+  }
+
+  type State = {
+    [k in string | number]: unknown;
+  }
+
+  interface RouteLocation {
+    pathname: string;
+    search: string;
+    hash?: string;
+    state: State;
+  }
+
+  interface RouteParams {
+    [param: string]: string;
+  }
+
+  interface LinkProps {
+    to: string;
```

Comment:
👏

Reply:
Special thanks to @mefechoel, who was of huge help 😄